src/autocompletion/completion-provider.ts

```diff
@@ -118,7 +118,7 @@ export class CompletionProvider implements vscode.CompletionItemProvider {
       if (token) {
         let interpolationCompletions = this.interpolationCompletions(document, position);
         if (interpolationCompletions.length !== 0)
-          return interpolationCompletions;
```

Comment:
unsure if this does what you want. If we really want this to be a unique list, rather than just ensure that the results have more information (e.g. show some sort of information where the resource is defined in case of duplicates), I think the making it unique part should be part of `interpolationCompletions` (e.g here: https://github.com/mauve/vscode-terraform/pull/100/commits/764bd8d4117ea943f3497e12816d7832786149a0#diff-c8cbd71ada55b23b86d5b1e2aac1255eL101)

Uniqueness of `Segment`s (which is the things returned by `(File)Index::query()` can be ensured using `Segment::id()`. 

In any case I would prefer if we could also add a test for this in https://github.com/mauve/vscode-terraform/blob/master/test/autocompletion/completion-provider.test.ts or in any of the test files in https://github.com/mauve/vscode-terraform/tree/master/test/index depending on where the uniqueness is guaranteed.